tools/generate_tables_sql_and_testNames.php

```diff
@@ -1,118 +1,234 @@
+                $sql_query_columns[] = "`Data_entry_completion_status` "
+                    ."enum('Incomplete','Complete') "
+                    ."NOT NULL default 'Incomplete',\n";
+                break;
             case "page":
                 $pages[] = $bits[2];
-                continue;
-
-            //no SQL need be generated.
+                continue 2;
+                // Continue as no SQL needs to be generated.
             case "title":
                 $title = $bits[1];
             case "header":
                 continue;
-            break;
-
-            //generate specific column definitions for specific types of HTML elements
+                break;
+                // Generate specific column definitions for types of HTML elements
             default:
-                if($bits[1] == "") {
+                if ($bits[1] == "") {
                     continue;
                 }
-                if($bits[0]=="select"){
-                    $bits[0]=enumizeOptions($bits[3], isset($tablename) ? $tablename : null, $bits[1]);
-                } else if($bits[0]=="selectmultiple"){
-                    $bits[0]="varchar(255)";
-                } else if($bits[0]=="textarea"){
-                    $bits[0]="text";
-                } else if($bits[0]=="text"){
-                    $bits[0]="varchar(255)";
-                } else if($bits[0]=="checkbox") {
-                    $bits[0]="varchar(255)";
-                } else if ($bits[0]=="static") {
-                    $bits[0]="varchar(255)";
+                if ($bits[0] == "select") {
+                    $bits[0] = enumizeOptions($bits[3]);
+                } else if ($bits[0] == "selectmultiple") {
+                    $bits[0] = "varchar(255)";
+                } else if ($bits[0] == "textarea") {
+                    $bits[0] = "text";
+                } else if ($bits[0] == "text") {
+                    $bits[0] = "varchar(255)";
+                } else if ($bits[0] == "checkbox") {
+                    $bits[0] = "varchar(255)";
+                } else if ($bits[0] == "static") {
+                    $bits[0] = "varchar(255)";
+                } else {
+                    // Sanitize $bits to mitigate SQL injection attacks.
+                    $bits[0] = str_replace('`', '', $db->escape($bits[0]));
                 }
-
-                $bits[2]=htmlspecialchars($bits[2]);
-                $output.="`$bits[1]` $bits[0] default NULL,\n";
+                $bits[2]     = htmlspecialchars($bits[2]);
+                $output     .= "`$sql_query_table_name` $bits[0] default NULL,\n";
+                $column_name = str_replace('`', '', $db->escape($bits[1]));
+                $sql_query_columns[] = "`$column_name` $bits[0] default NULL,\n";
+            }
         }
-
+        $output .= "PRIMARY KEY  (`CommentID`)\n) ENGINE=InnoDB"
+            ." DEFAULT CHARSET=utf8;\n";
+        $fp      = fopen($file_name, "w");
```

Comment:
do we want to write the file if we are using the --apply ?

Reply:
I haven't seen --apply before.